src/OpenTelemetry.Collector.Dependencies/SqlClientCollector.cs

```diff
@@ -0,0 +1,48 @@
+﻿// <copyright file="SqlClientCollector.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System;
+using OpenTelemetry.Collector.Dependencies.Implementation;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Collector.Dependencies
+{
+    /// <summary>
+    /// SqlClient collector.
+    /// </summary>
+    public class SqlClientCollector : IDisposable
+    {
+        private readonly DiagnosticSourceSubscriber diagnosticSourceSubscriber;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SqlClientCollector"/> class.
+        /// </summary>
+        /// <param name="tracer">Tracer to record traced with.</param>
+        public SqlClientCollector(ITracer tracer)
+        {
+            this.diagnosticSourceSubscriber = new DiagnosticSourceSubscriber(
+               name => new SqlClientDiagnosticListener(name, tracer),
+               listener => listener.Name.Contains("SqlClient"),
```

Comment:
please create listener per source ( one for System.Data, another one for Ms.Data) it will simplify fetchers managemenent a LOT

Reply:
They both have the same name for a listener - `SqlClientDiagnosticListener`. There is no way to detect - what version of SqlClient is using.

Reply:
ah, as long as you use ` name => new SqlClientDiagnosticListener(name, tracer),` it will create a listener per source, i.e. we are good here.

Then please use the exact match with source name - `SqlClientDiagnosticListener` there is the way it can work with any abstract listener except `SqlClientDiagnosticListener`.